src/features/user/Widget/DonationLink/DonationLinkForm.tsx

```diff
@@ -0,0 +1,244 @@
+  const [isProjectSelected, setIsProjectSelected] = useState<boolean>(false);
+  // Load all projects
+
+  const handleChange = () => {
+    const link = isTesting
+      ? 'http://paydev.pp.eco/'
+      : process.env.NEXT_PUBLIC_DONATION_URL;
+    const url = `${link}?country=${country}&locale=${Languages.langCode}${
+      localProject == null
+        ? ''
+        : `&to=${localProject.name.split(/\W+/).join('-').toLowerCase()}`
+    }&tenant=${TENANT_ID}${isSupport ? `&s=${user.slug}` : ''}
+    `;
+    setDonationUrl(url);
+  };
+
+  const handleProjectChange = async (project: Project | null) => {
+    setLocalProject(project);
+    setIsProjectSelected(!isProjectSelected);
+  };
+
+  useEffect(() => {
+    handleChange();
+  }, [country, Languages, localProject, isSupport, isTesting]);
+
+  // const setTextCopiedClipboard = () => {
+  //   navigator.clipboard.writeText(donationUrl);
+  //   setIsCopied(!isCopied);
+  // };
+
+  if (ready) {
+    return (
+      <StyledForm>
+        <div className="inputContainer">
+          <div className={styles.singleFormGroup}>
+            <div className={styles.formHeader}>
+              {t('donationLink:countryLanguageTitle')}
+            </div>
+            <InlineFormDisplayGroup>
+              <AutoCompleteCountry
+                label={t('donationLink:labelCountry')}
+                name="country"
+                defaultValue={country}
+                onChange={setCountry}
+              />
+              <MuiAutocomplete
+                id={t('donationLink:labelLanguages')}
+                options={supportedLanguages}
+                getOptionLabel={(option) =>
+                  `${option.langCode} - ${option.languageName}`
+                }
+                isOptionEqualToValue={(option, value) =>
+                  option.langCode === value.langCode
+                }
+                value={Languages}
+                renderInput={(params) => (
+                  <TextField
+                    {...params}
+                    label="Language"
+                    placeholder={t('donationLink:Languages')}
+                  />
+                )}
+                renderOption={(props, option) => (
+                  <StyledAutoCompleteOption {...props} key={option.langCode}>
+                    <span>{`${option.langCode}`}</span>
+                    {`- ${option.languageName}`}
+                  </StyledAutoCompleteOption>
+                )}
+                onChange={(event, newLan) => setLanguage(newLan)}
+              />
+            </InlineFormDisplayGroup>
+          </div>
+          <div className={styles.singleFormGroup}>
+            <div className={styles.formHeader}>
+              {t('donationLink:projectTitle')}
+            </div>
+            <ProjectSelectAutocomplete
+              handleProjectChange={handleProjectChange}
+              project={localProject}
+              projectList={projectsList || []}
+              active={true}
+            />
```

Comment:
There's a console warning that appears the first time the user tries to select a project.

`Warning: Each child in a list should have a unique "key" prop.`

Reply:
Fixed